apps/examples/mediaplayer/BufferInputDataSource.cpp

```diff
@@ -36,13 +36,13 @@ bool BufferInputDataSource::isPrepared()
 
 bool BufferInputDataSource::open()
 {
-	mFp = fopen("/rom/over_16000.mp3", "rb");
-	mSrcSize = 2048;
-	mSrcBuf = new unsigned char[mSrcSize];
-	setSampleRate(16000);
-	setChannels(2);
-	setPcmFormat(media::AUDIO_FORMAT_TYPE_S16_LE);
-	setAudioType(media::AUDIO_TYPE_MP3);
+	if (!isPrepared()) {
+		mFp = fopen("/rom/over_16000.mp3", "rb");
+		mSrcSize = 2048;
+		mSrcBuf = new unsigned char[mSrcSize];
+		// Application can specify all audio informations (audioType/sampleRate/channels/pcmFormat),
+		// Otherwise, player will identify them from data stream during prepareAsync().
+	}
 	return mFp && mSrcBuf && mSrcSize;
```

Comment:
How about replacing this part with isPrepared?

Reply:
good, I will change it.